source/common/upstream/cluster_manager_impl.cc

```diff
@@ -121,15 +134,16 @@ void ClusterManagerInitHelper::maybeFinishInitialize() {
   if (!secondary_init_clusters_.empty()) {
     if (!started_secondary_initialize_) {
       ENVOY_LOG(info, "cm init: initializing secondary clusters");
-      started_secondary_initialize_ = true;
-      // Cluster::initialize() method can modify the list of secondary_init_clusters_ to remove
-      // the item currently being initialized, so we eschew range-based-for and do this complicated
-      // dance to increment the iterator before calling initialize.
-      for (auto iter = secondary_init_clusters_.begin(); iter != secondary_init_clusters_.end();) {
-        Cluster* cluster = *iter;
-        ++iter;
-        ENVOY_LOG(debug, "initializing secondary cluster {}", cluster->info()->name());
-        cluster->initialize([cluster, this] { onClusterInit(*cluster); });
+      // If the first CDS response doesn't have any primary cluster, ClusterLoadAssignment
```

Comment:
I am little confused with this comment. Don't we pause `ClusterLoadAssignment`, irrespective of whether we have primary or secondary clusters here https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/cds_api_impl.cc#L59?

Reply:
Ok. Thanks. That helps. This could happen if there are primary clusters in static boostrap as well I guess, not just CDS's first response?